test/Microsoft.Sbom.Targets.E2E.Tests/GenerateSbomE2ETests.cs

```diff
@@ -0,0 +1,344 @@
+        }
+    }
+
+    private void RestoreBuildPack(Project sampleProject)
+    {
+        var logger = new ConsoleLogger();
+
+        // Restore the project to create project.assets.json file
+        var restore = sampleProject.Build("Restore", new[] { logger });
+        Assert.IsTrue(restore, "Failed to restore the project");
+
+        // Next, build the project
+        var build = sampleProject.Build(logger);
+        Assert.IsTrue(build, "Failed to build the project");
+
+        // Finally, pack the project
+        var pack = sampleProject.Build("Pack", new[] { logger });
+        Assert.IsTrue(pack, "Failed to pack the project");
+    }
+
+    private void ExtractPackage()
+    {
+        // Unzip the contents of the NuGet package
+        var nupkgPath = Path.Combine(projectDirectory, "bin", configuration);
+        var nupkgFile = Path.Combine(nupkgPath, $"{expectedPackageName}.{expectedVersion}.nupkg");
+        var zipFile = Path.Combine(nupkgPath, $"{expectedPackageName}.{expectedVersion}.zip");
+        var extractPath = Path.Combine(projectDirectory, "bin", configuration, $"{Guid.NewGuid()}.temp");
+
+        // Rename the .nupkg file to .zip
+        File.Copy(nupkgFile, zipFile, true);
+
+        // Extract the .zip file
+        ZipFile.ExtractToDirectory(zipFile, extractPath);
+
+        manifestPath = Path.Combine(extractPath, "_manifest", sbomSpecificationDirectoryName, "manifest.spdx.json");
+    }
+
+    [TestMethod]
+    public void SbomGenerationSucceedsForDefaultProperties()
+    {
+        if (!IsWindows)
+        {
+            Assert.Inconclusive("This test is not (yet) supported on non-Windows platforms.");
+            return;
+        }
+
+        // Create and setup a Project object for ProjectSample1
+        var sampleProject = SetupSampleProject();
+
+        // Restore, build, and pack the project
+        RestoreBuildPack(sampleProject);
+
+        // Extract the NuGet package
+        ExtractPackage();
+
+        // Validate the SBOM exists in the package.
+        Assert.IsTrue(File.Exists(manifestPath));
+    }
+
+    [TestMethod]
+    public void SbomGenerationSucceedsForValidNamespaceBaseUriUniquePart()
+    {
+        if (!IsWindows)
+        {
+            Assert.Inconclusive("This test is not (yet) supported on non-Windows platforms.");
+            return;
+        }
+
+        // Create and setup a Project object for ProjectSample1
+        var sampleProject = SetupSampleProject();
+
+        // Manually set the NamespaceUriUniquePart
+        var namespaceUriUniquePart = Guid.NewGuid().ToString();
+        sampleProject.SetProperty("SbomGenerationNamespaceUriUniquePart", namespaceUriUniquePart);
+
+        // Restore, build, and pack the project
+        RestoreBuildPack(sampleProject);
+
+        // Extract the NuGet package
+        ExtractPackage();
+
+        // Validate the SBOM exists in the package.
+        Assert.IsTrue(File.Exists(manifestPath));
```

Comment:
Since we do a more thorough validation for the regular unit tests (through our GeneratedSbomValidator class), we thought it would be fine just verifying the existence of the SBOM for the E2E tests. Should we opt for more in depth validation for the E2E tests as well?

Reply:
That's fine with me!